docs/references/api/resource_embedding.rst

```diff
@@ -318,10 +320,294 @@ Computed relationships have good performance as their intended design enable `in
 
+.. _embed_disamb:
+.. _hint_disamb:
+.. _target_disamb:
+.. _complex_rels:
+
+Complex Relationships
+=====================
+
+As mentioned on :ref:`resource_embedding`, the server does joins based on **Foreign Keys** columns. When there are many foreign keys between tables,
+it needs disambiguation to resolve which foreign key columns to use for the join.
+
+:ref:`computed_relationships` can do the job here, they can choose join columns arbitrarily.
+
+.. _multiple_m2o:
+
+Multiple Many-To-One
+--------------------
+
+.. tabs::
+
+  .. group-tab:: ERD
+
+    .. image:: ../../_static/orders.png
+
+  .. code-tab:: postgresql SQL
+
+    create table addresses (
+      id int primary key generated always as identity,
+      name text,
+      city text,
+      state text,
+      postal_code char(5)
+    );
+
+    create table orders (
+      id int primary key generated always as identity,
+      name text,
+      billing_address_id int references addresses(id),
+      shipping_address_id int references addresses(id)
+    );
+
+To successfully embed ``orders`` with ``addresses``, you need to create computed relationships for the foreign keys columns you want to use:
+
+.. code-block:: postgresql
+
+  create function billing_address(orders) returns setof addresses rows 1 as $$
+    select * from addresses where id = $1.billing_address_id
+  $$ stable language sql;
+
+  create function shipping_address(orders) returns setof addresses rows 1 as $$
+    select * from addresses where id = $1.shipping_address_id
+  $$ stable language sql;
+
+Now we can unambiguously embed the billing address by specifying the ``billing_address`` computed relationship.
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /orders?select=name,billing_address(name) HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/orders?select=name,billing_address(name)"
+
+.. code-block:: json
+
+   [
+    {
+     "name": "Personal Water Filter",
+     "billing_address": {
+       "name": "32 Glenlake Dr.Dearborn, MI 48124"
+     }
+    }
+   ]
+
+.. _recursive_o2o_embed:
+
+Recursive One-To-One
+--------------------
+
+.. tabs::
+
+  .. group-tab:: ERD
+
+    .. image:: ../../_static/presidents.png
+
+  .. code-tab:: postgresql SQL
+
+    create table presidents (
+      id int primary key generated always as identity,
+      first_name text,
+      last_name text,
+      predecessor_id int references presidents(id) unique
+    );
+
+To get either side of the Recursive One-To-One relationship, create the functions:
+
+.. code-block:: postgresql
+
+  create or replace function predecessor(presidents) returns setof presidents rows 1 as $$
+    select * from presidents where id = $1.predecessor_id
+  $$ stable language sql;
+
+  create or replace function successor(presidents) returns setof presidents rows 1 as $$
+    select * from presidents where predecessor_id = $1.id
+  $$ stable language sql;
```

Comment:
For completeness, let's also use `successor` in the request. It's fine to use both on a single one.